src/TypeFormatter/TupleTypeFormatter.ts

```diff
@@ -20,13 +20,29 @@ export class TupleTypeFormatter implements SubTypeFormatter {
         const optionalElements = subTypes.filter((t) => t instanceof OptionalType) as OptionalType[];
         const restElements = subTypes.filter((t) => t instanceof RestType) as RestType[];
 
+        const restType = restElements.length ? restElements[0].getType().getItem() : undefined;
+        const restDefinition = restType ? this.childTypeFormatter.getDefinition(restType) : undefined;
```

Comment:
Let's move this into the conditional below since we don't need it if the tuple is not uniform

Reply:
Not sure I follow, if I move it down I'll have to expand it in the new code.

Reply:
If the tuple is not uniform you might still need it, for example in `[A, B, ...A[]]`.

Reply:
`restDefinition` is only used if `isUniformArray` is true so you can move `restDefinition` into the if block below.

Reply:
It's also used if `isUniformArray` is false, in the last four lines of the other return value.

Reply:
Ah, I missed that in the diff.